[RM Ch07] Add Security Requirements (#1118)
cntt-n/CNTT: doc/ref_model/chapters/chapter07.md

    tools
-   Resiliency tests run (such as hardware failures or power failure
    tests)

<a name="7.11"></a>
## 7.11 Consolidated Security Requirements

> Please note that **Virtual Resource Manager (VRM)** is used to collectively represent VIM, K8s, …

Reviewer: Can you complete this sentence ```to collectively represent VIM, K8s, …```

Author: @msidana Done - Thanks

<a name="7.11.1"></a>
### 7.11.1. System Hardening

|  Ref | Requirement  |  Unit  | Definition  |
|-------|------|------|-------|
| sec.gen.001 | The Platform **must** maintain the state to what it is specified to be and does not change unless through change management process. |   |  |
| sec.gen.002 | All systems part of NFVI **must** support password hardening (strength and rules for updates (process), storage and transmission, etc.) |  | Hardening: NIST SP 800-63B |

> | sec.gen.003 | All servers part of NFVI **must** support secure boot, root of trust and trusted boot |  | Secure boot - measured boot, verified boot  <br/> Root of trust: specified in RA or RI; should CNTT specify how? |

Author: @glimonta Thanks. Reworded sec.gen.003 as you suggested and added sec.gen.014 with your second requirement. Deleted my comment (question) from sec.gen.003.  Kindly review.

> | sec.gen.003 | All servers part of NFVI **must** support secure boot, root of trust and trusted boot |  | Secure boot - measured boot, verified boot  <br/> Root of trust: specified in RA or RI; should CNTT specify how? |

Reviewer 2: Looks good. Thanks.

| sec.gen.004 | The Operating Systems of all the servers part of NFVI **must** be hardened |  | NIST SP 800-123 |
| sec.gen.005 | The Platform **must** support Operating System level access control |  | Details on OS |
| sec.gen.006 | The Platform **must** support Secure logging |  | Details |
| sec.gen.007 | All servers part of NFVI **must** be Time synchronized with authenticated Time service |  | |
| sec.gen.008 | All servers part of NFVI **must** be regularly updated to address security vulnerabilities |  | |
| sec.gen.009 | The Platform **must** support Software integrity protection and verification |  | |
| sec.gen.010 | The NFVI **must** support Secure storage (all types) |  | Expand/Delete based on other reqts |
| sec.gen.011 | The NFVI **should** support Read and Write only storage partitions (write only permission to one or more authorized actors) |  | NIST reference? |
| sec.gen.012 | The Platform **must** ensure that only authorized actors have physical access to the underlying infrastructure. |  | |

> | sec.gen.013 | The Platform **must** ensure that only authorized actors have logical access to the underlying infrastructure. |  | |

Reviewer: What do you mean by logical access ? Do you intend SSH etc access by this requirement ?

Author: @msidana Yes that is what is in the NIST doc

<a name="7.11.21"></a>
###  7.11.2. Platform Access

Ref | Requirement | Unit | Definition |
|-------|-------|-------|---------|
| sec.sys.001 | The Platform **must** support authenticated and secure APIs, API endpoints |  | |
| | The Platform **must** implement authenticated and secure access to GUI |  | |
| sec.sys.002 | The Platform **must** support Traffic Filtering for workloads (for example, Fire Wall) |  | |
| sec.sys.003 | The Platform **must** support Secure and encrypted communications, and confidentiality and integrity of network traffic |  | |
| sec.sys.004 | The NFVI **must** support Secure network channels |  | A secure channel enables transferring of data that is resistant to overhearing and tampering |
| sec.sys.005 | The NFVI **must** segregate the underlay and overlay networks |  | |

> | sec.sys.006 | The NFVI must be able to utilize the VRM identity management capabilities |  | |

Reviewer: How can NFVI utilise VRM (ex VIM) identity management capabilities?

Author: @msidana maybe needs an explanation but the NFVI includes the VMs (and hence the workloads) and they should be able to utilise the appropriate Identity service APIs.

Reviewer: Hmm. VM's in this context is very generic. We should do some re-phrasing if we mean workloads - ```Workloads may be able to utilise VRM identity capabilities depending on the need of running application```.
++ I would also propose ```may``` here

Reviewer 3: @msidana is this change considered optional or mandatory?

Reviewer: Optional, according to me.

| sec.ci.008 | The NFVI **must** support tenant networks segregation |  | |

<a name="7.11.4"></a>
### 7.11.4. Workload Security

| Ref | Requirement | Unit | Definition |
|---|----|---|----|
| sec.wl.001 | The Platform **must** support Workload placement policy |  | |
| sec.wl.002 | The Platform **must** support operational security |  | |
| sec.wl.003 | The Platform **must** support secure provision of workloads  |  | |
| sec.wl.004 | The Platform **must** support Location assertion (for mandated in-country or location requirements) |  | |
| sec.wl.005 | Production workloads **must** be separated from non-production workloads |  | |
| sec.wl.006 | Workloads **must** be separable by sensitivity level (for example, financial versus external facing web server) |  | |
| sec.wl.007 | Platform **must** verify VNF authenticity and integrity |  | |

<a name="7.11.5"></a>
### 7.11.5. Image Security

| Ref | Requirement | Unit | Definition |
|---|----|---|----|

> | sec.img.001 | Only images from trusted sources **must** be used |  | |

Reviewer 3: ```suggestion
| sec.img.001 | Images from untrusted sources **must not** be used |  | |
```
A slight re-wording that I think clarifies the intent.

Author: @kedmison Done - Thanks

| Ref | Requirement | Unit | Definition |
|---|----|---|----|
| sec.mon.001 | Platform must provide logs and these logs must be regularly scanned for events of interest |  | |
| sec.mon.002 | Security logs must be time synchronised |  | |
| sec.mon.003 | The Platform must log all changes to time server source, time, date and time zones |  | |
| sec.mon.004 | The Platform must secure and protect Audit logs (contain sensitive information) both in-transit and at rest |  | |
| sec.mon.005 | The Platform must Monitor and Audit various behaviours of connection and login attempts to detect access attacks and potential access attempts and take corrective actions accordingly |  | |
| sec.mon.006 | The Platform must Monitor and Audit operations by authorized account access after login to detect malicious operational activity and take corrective actions accordingly |  | |
| sec.mon.007 | The Platform must Monitor and Audit security parameter configurations for compliance with defined security policies |  | |
| sec.mon.008 | The Platform must Monitor and Audit externally exposed interfaces for illegal access (attacks) and take corrective security hardening measures |  | |

> | sec.mon.009 | The Platform must Monitor and Audit service handling for various attacks (malformed messages, signalling flooding and replaying, etc.) and take corrective actions accordingly |  | |

Reviewer 3: "take corrective actions accordingly" is not a testable requirement: can this be something like raise an alert or notification?

> | sec.mon.010 | The Platform must Monitor and Audit running processes to detect unexpected or unauthorized processes and take corrective actions accordingly |  | |

Reviewer 3: "take corrective actions accordingly" is not a testable requirement: can this be something like raise an alert or notification?

> | sec.mon.011 | The Platform must Monitor and Audit logs from infrastructure elements and workloads to detected anomalies in the system components and take corrective actions accordingly |  | |

Reviewer 3: "take corrective actions accordingly" is not a testable requirement: can this be something like raise an alert or notification?

> | sec.mon.012 | The Platform must Monitor and Audit Traffic patterns and volumes to detect inbound or outbound DoS attacks, and to prevent malware download attempts |  | |

Reviewer 3: ```suggestion
```
DDOS detection and prevention is out of scope for NFVI definition.

Author: @kedmison removed DOS

> | sec.mon.013 | The Security monitoring system must not affect the security (integrity and confidentiality) of the infrastructure, workloads, or the user data (through back door entries). |  | |

Reviewer 3: ```suggestion
| sec.mon.013 | The monitoring system must not affect the security (integrity and confidentiality) of the infrastructure, workloads, or the user data (through back door entries). |  | |
```
Removing the mention of 'security' monitoring system as that suggests a whole new system, separate from the log monitoring.

Author: @kedmison Done -- Thanks

| sec.mon.014 | The Monitoring systems should not impact IAAS, PAAS, and SAAS SLAs including availability SLAs |  | |

> | sec.mon.015 | The Platform must ensure that the Security Monitoring systems are never starved of resources |  | |

Reviewer 3: ```suggestion
| sec.mon.015 | The Platform must ensure that the monitoring systems are never starved of resources |  | |
```

Author: @kedmison Done - Thanks

> | sec.mon.016 | The Platform Security Monitoring components should follow security best practices for auditing, including secure logging and tracing |  | |

Reviewer 3: ```suggestion
| sec.mon.016 | The Platform monitoring components should follow security best practices for auditing, including secure logging and tracing |  | |
```

Author: @kedmison Done - Thanks

    tools
-   Resiliency tests run (such as hardware failures or power failure
    tests)

<a name="7.11"></a>
## 7.11 Consolidated Security Requirements

> Please note that **Virtual Resource Manager (VRM)** is used to collectively represent VIM, K8s, etc.

Reviewer: I think the definition of VRM should be clear, as in what it includes. Using ```etc.``` is open-ended.

Author: @msidana Given the move away from NFVI (will edit), VRM will be changed to "Cloud Infrastructure Manager (CIM)"

Author: @msidana Changes made

<a name="7.11.1"></a>
### 7.11.1. System Hardening

|  Ref | Requirement  |  Unit  | Definition  |
|-------|------|------|-------|
| sec.gen.001 | The Platform **must** maintain the state to what it is specified to be and does not change unless through change management process. |   |  |
| sec.gen.002 | All systems part of NFVI **must** support password hardening (strength and rules for updates (process), storage and transmission, etc.) |  | Hardening: NIST SP 800-63B |
| sec.gen.003 | All servers part of NFVI **must** support a root of trust and secure boot |  |  |
| sec.gen.004 | The Operating Systems of all the servers part of NFVI **must** be hardened |  | NIST SP 800-123 |
| sec.gen.005 | The Platform **must** support Operating System level access control |  | Details on OS |
| sec.gen.006 | The Platform **must** support Secure logging |  | Details |
| sec.gen.007 | All servers part of NFVI **must** be Time synchronized with authenticated Time service |  | |
| sec.gen.008 | All servers part of NFVI **must** be regularly updated to address security vulnerabilities |  | |
| sec.gen.009 | The Platform **must** support Software integrity protection and verification |  | |

> | sec.gen.010 | The NFVI **must** support Secure storage (all types) |  | Expand/Delete based on other reqts |

Reviewer: Pls expand reqts.

Author: @msidana Done - Thanks

> | sec.gen.011 | The NFVI **should** support Read and Write only storage partitions (write only permission to one or more authorized actors) |  | NIST reference? |

Reviewer: write only -> write-only. Pls complete this NIST reference

Author: @msidana It was decided not to include US references -- sorry had missed this.

> | sec.gen.012 | The Operator **must** ensure that only authorized actors have physical access to the underlying infrastructure. |  | |

Reviewer: sec.gen.012 is a physical aspect related to access. Does it needs to be a part of RM ?

Author: @msidana In terms of security it is critically important and is equally applicable to all implementations. Hence, the decision to include after some discussion.

| sec.gen.013 | The Platform **must** ensure that only authorized actors have logical access to the underlying infrastructure. |  | |
| sec.gen.014 | All servers part of NFVI **should** support measured boot and an attestation server that monitors the measurements of the servers. |  | |

<a name="7.11.21"></a>
###  7.11.2. Platform Access

Ref | Requirement | Unit | Definition |
|-------|-------|-------|---------|
| sec.sys.001 | The Platform **must** support authenticated and secure APIs, API endpoints |  | |
| | The Platform **must** implement authenticated and secure access to GUI |  | |
| sec.sys.002 | The Platform **must** support Traffic Filtering for workloads (for example, Fire Wall) |  | |
| sec.sys.003 | The Platform **must** support Secure and encrypted communications, and confidentiality and integrity of network traffic |  | |

> | sec.sys.004 | The NFVI **must** support Secure network channels |  | A secure channel enables transferring of data that is resistant to overhearing and tampering |

Reviewer: Why have you kept NFVI security requirements under Platform Access? Any specific reason?

Author: @msidana Changing the Title to Platform and Access. Thanks.

| sec.ci.005 | The Platform **must** support Confidentiality and Integrity of process-related metadata and restrict information sharing with only the process owner (e.g., tenant). |  | |
| sec.ci.006 | The Platform **must** support Confidentiality and Integrity of workload resource utilization (RAM, CPU, Storage, Network I/O, cache, hardware offload) and restrict information sharing with only the workload owner (e.g., tenant). |  | |
| sec.ci.007 | The Platform **must not** allow Memory Inspection by any actor other than the authorized actors for the Entity to which Memory is assigned (e.g., tenants owning the workload), for Lawful Inspection, and by secure monitoring services |  | Admin access must be carefully regulated |
| sec.ci.008 | The NFVI **must** support tenant networks segregation |  | |

<a name="7.11.4"></a>
### 7.11.4. Workload Security

| Ref | Requirement | Unit | Definition |
|---|----|---|----|
| sec.wl.001 | The Platform **must** support Workload placement policy |  | |
| sec.wl.002 | The Platform **must** support operational security |  | |

> | sec.wl.003 | The Platform **must** support secure provision of workloads  |  | |

Reviewer: provision -> provisioning.

| sec.gen.003 | All servers part of NFVI **must** support secure boot, root of trust and trusted boot |  | Secure boot - measured boot, verified boot  <br/> Root of trust: specified in RA or RI; should CNTT specify how? |
| sec.gen.004 | The Operating Systems of all the servers part of NFVI **must** be hardened |  | NIST SP 800-123 |
| sec.gen.005 | The Platform **must** support Operating System level access control |  | Details on OS |
| sec.gen.006 | The Platform **must** support Secure logging |  | Details |
| sec.gen.007 | All servers part of NFVI **must** be Time synchronized with authenticated Time service |  | |
| sec.gen.008 | All servers part of NFVI **must** be regularly updated to address security vulnerabilities |  | |
| sec.gen.009 | The Platform **must** support Software integrity protection and verification |  | |
| sec.gen.010 | The NFVI **must** support Secure storage (all types) |  | Expand/Delete based on other reqts |
| sec.gen.011 | The NFVI **should** support Read and Write only storage partitions (write only permission to one or more authorized actors) |  | NIST reference? |

> | sec.gen.012 | The Platform **must** ensure that only authorized actors have physical access to the underlying infrastructure. |  | |

Reviewer 4: How can a Platfrm ensure this?

Author: @CsatariGergely Platform = NFVI + VRM (such as VIM). Please suggest alternate language if the explanation is unsatisfactory.

Reviewer 3: The platform does not encompass the building or facilities, server cages, etc needed to deny physical access to the servers.  This is a valid datacenter requirement, but it should not be a CNTT requirement.  I suggest removing sec.gen.012 as it is out of scope for CNTT and thus unenforceable.
```suggestion
```

Reviewer 3: I think sec.gen.012 is a datacenter requirement, not a platform requirement.  If the platform is limited to the servers, storage, networking, and software used to create the cloud infrastructure it doesn't incorporate physical datacenter design aspects (like physical security).  A vendor NFVI cannot be deemed to comply with this requirement as it needs to be installed in a physical location that is not of the vendor's choosing.
Can this requirement be removed?  I know it's a valid requirement from a holistic perspective, but when considering the Cloud Infrastructure, it's not in that area of focus and thus should be specified in other documents but not CNTT ones.

| sec.gen.013 | The Platform **must** ensure that only authorized actors have logical access to the underlying infrastructure. |  | |

<a name="7.11.21"></a>
###  7.11.2. Platform Access

Ref | Requirement | Unit | Definition |
|-------|-------|-------|---------|
| sec.sys.001 | The Platform **must** support authenticated and secure APIs, API endpoints |  | |
| | The Platform **must** implement authenticated and secure access to GUI |  | |
| sec.sys.002 | The Platform **must** support Traffic Filtering for workloads (for example, Fire Wall) |  | |
| sec.sys.003 | The Platform **must** support Secure and encrypted communications, and confidentiality and integrity of network traffic |  | |
| sec.sys.004 | The NFVI **must** support Secure network channels |  | A secure channel enables transferring of data that is resistant to overhearing and tampering |
| sec.sys.005 | The NFVI **must** segregate the underlay and overlay networks |  | |

> | *sec.sys.006* | *Placeholder* |  | |

Reviewer 4: What is this?

Author: @CsatariGergely A requirement was deleted during the review -- renumbering will occur once we have all issues addressed. Hope that is OK.

Author: @CsatariGergely Deleted and list renumbered.

| sec.ci.006 | The Platform **must** support Confidentiality and Integrity of workload resource utilization (RAM, CPU, Storage, Network I/O, cache, hardware offload) and restrict information sharing with only the workload owner (e.g., tenant). |  | |
| sec.ci.007 | The Platform **must not** allow Memory Inspection by any actor other than the authorized actors for the Entity to which Memory is assigned (e.g., tenants owning the workload), for Lawful Inspection, and by secure monitoring services |  | Admin access must be carefully regulated |
| sec.ci.008 | The NFVI **must** support tenant networks segregation |  | |

<a name="7.11.4"></a>
### 7.11.4. Workload Security

| Ref | Requirement | Unit | Definition |
|---|----|---|----|
| sec.wl.001 | The Platform **must** support Workload placement policy |  | |
| sec.wl.002 | The Platform **must** support operational security |  | |
| sec.wl.003 | The Platform **must** support secure provision of workloads  |  | |
| sec.wl.004 | The Platform **must** support Location assertion (for mandated in-country or location requirements) |  | |

> | sec.wl.005 | Production workloads **must** be separated from non-production workloads |  | |

Author: @kedmison Solutions include separate AZs or host aggregates (HA) where the non-prod HA doesn't share hosts (servers) with prod HAs

> | sec.wl.005 | Production workloads **must** be separated from non-production workloads |  | |

Reviewer 3: I can see AZs being used for this.  Host aggregates I'm less clear on how that would be expressed, at run-time, without changing (for RA1) Openstack flavour definitions or adding extra criteria at VM instantiation time.

> | sec.wl.006 | Workloads **must** be separable by sensitivity level (for example, financial versus external facing web server) |  | |

Author: @kedmison Maybe for CNTT with only VNF workloads this is unnecessary. However, in many places within the RM, IT and VNF workloads are mentioned. In the case of IT workloads it is quite common to classify them and hence require different levels of treatment.

> | sec.wl.006 | Workloads **must** be separable by sensitivity level (for example, financial versus external facing web server) |  | |

Reviewer 3: Would one use Availability Zones again for this sort of workload separation?

> | sec.wl.006 | Workloads **must** be separable by sensitivity level (for example, financial versus external facing web server) |  | |

Author: @kedmison We do not want to suggest a solution but in OSTK AZ or even a Region, and in container-world a different cluster ...

> | sec.wl.006 | Workloads **must** be separable by sensitivity level (for example, financial versus external facing web server) |  | |

Reviewer 3: I think we need to drill into this use case more; I don't see a testable definition of separable here.
(e.g. is a VM boundary sufficient?  does it have to be a host boundary?  host aggregate? AZ? Datacenter?  container?). Each of those is a separation; but from our discussion thread, it seems that only some would qualify as 'separation' for this requirement.

> | sec.wl.006 | Workloads **must** be separable by sensitivity level (for example, financial versus external facing web server) |  | |

Author: @kedmison the requirement has been reworded to "Workloads must be separable by their categorisation (for example, payment card information, healthcare, etc.)" -- does that work?

> | sec.wl.007 | Platform **must** verify VNF authenticity and integrity |  | |

Author: @kedmison Please suggest where you would like this to be addressed

> | sec.wl.007 | Platform **must** verify VNF authenticity and integrity |  | |

Reviewer 3: I think this should be a fully separate Github issue of its own, with some serious discussion and light brought to it, due to the complexity and diversity of impact.

> | sec.wl.007 | Platform **must** verify VNF authenticity and integrity |  | |

Reviewer 3: ```suggestion
```
Suggest striking this requirement until we can decide on what the VNF-NFVI interface would be for the authenticity and integrity, and what the impacts on VNF vendors would be at build time, etc.

<a name="7.11.5"></a>

> ### 7.11.5. Image Security

Author: @kedmison Did you mean RM Appendix A: VNF Design Guidelines?

> ### 7.11.5. Image Security

Reviewer 3: Yes, I meant RM Appendix A.  Its title in the repo is "Guidelines for VNF Vendors".  https://github.com/cntt-n/CNTT/blob/master/doc/ref_model/chapters/appendix-a.md
The title used to be VNF design guidelines but changed some time ago.

> ### 7.11.5. Image Security

Author: @kedmison From you list, I think sec.img.001"Only images from trusted sources **must** be used" and sec.img.002 "Images **must** be maintained to be free from known vulnerabilities" are operator responsibilities.

Deleting sec.img.003-006 and sec.img.008 from this chapter and will open an issue to add to RM Appendix A.

| Ref | Requirement | Unit | Definition |
|---|----|---|----|
| sec.img.001 | Only images from trusted sources **must** be used |  | |
| sec.img.002 | Images **must** be maintained to be free from known vulnerabilities |  | |
| sec.img.003 | Images **must not** contain malicious code (e.g., malware, logic bombs, etc.)  |  | |
| sec.img.004 | Images **must not** contain code such as daemons that exposes them to risk |  | |
| sec.img.005 | Images **must** only be created with content and files from trusted sources |  | |
| sec.img.006 | Images **must** be packaged with files that have been found free of malware and vulnerabilities |  | |
| sec.img.007 | Images **must not** be configured to run with privileges higher than the privileges of the actor authorized to run them |  | |
| sec.img.008 | Images **must not** contain clear text secrets |  | |

> | sec.img.009 | Images **must** only be accessible to authorized actors |  | |

Author: @kedmison Yes - access/authorization testing for glance repos and even particular images

> | sec.img.009 | Images **must** only be accessible to authorized actors |  | |

Reviewer 3: This requirement may work for Glance (and not be a change) but it does represent a potential variance from standard process in the Kubernetes/container registry domain.  Can @tomkivlin and/or @CsatariGergely weigh in to see if this is acceptable for RA2?

| sec.img.012 | Image registries **must** be clear of vulnerable and stale (out of date) versions |  | |

<a name="7.11.6"></a>
### 7.11.6. Security LCM

| Ref | Requirement | Unit | Definition |
|---|----|---|----|
| sec.lcm.001 | The Platform **must** support Secure Provisioning, Maintaining availability, Deprovisioning (secure Clean-Up) of workload resources |  | Secure clean-up: tear-down, defending against virus or other attacks, or observing of cryptographic or user service data |
| sec.lcm.002 | Operational **must** use management protocols limiting security risk such as SNMPv3, SSH v2, ICMP, NTP, syslog and TLS |  | |
| sec.lcm.003 | The Cloud Operator **must** implement change management for NFVI, VRM and other components of the cloud |  | Platform change control on hardware |
| sec.lcm.004 | The Cloud Operator **should** support automated templated approved changes |  | Templated approved changes for automation where available |
| sec.lcm.005 | Platform **must** provide logs and these logs must be regularly scanned |  | |
| sec.lcm.006 | The Platform **must** verify the integrity of all Resource management requests | Yes/No | |

> | sec.lcm.007 | The Platform **must** be able to update newly instantiated, suspended, hibernated, migrated and restarted images with current time and time zone information |  | |

Reviewer 3: ```suggestion
| sec.lcm.007 | The Platform **must** be able to update newly instantiated, suspended, hibernated, migrated and restarted images with current time information |  | |
```
Time can be updated, as per the above comment.  Timezone must not be updated in the same fashion; guests and hosts should be independent when it comes to timezone.

> | sec.lcm.008 | The Platform **must** be able to update newly instantiated, suspended, hibernated, migrated and restarted images with relevant DNS information. |  | |

Reviewer 3: once a migrated or restarted workload is running, there is likely not an option to inject new DNS information without a planned co-ordination from the workload.  This means the workload is deciding when to look for, and/or obtain, updated DNS information  So, I am concerned that this requirement is not accurate wrt. normal timelines for the workload obtaining DNS information from the platform.

> | sec.lcm.009 | The Platform **must** be able to update newly instantiated, suspended, hibernated, migrated and restarted images with relevant IP addressing, including routing tables, route health information and whitelists/blacklists. |  | |

Author: @kedmison The fixed IP address assigned to an instance doesn't change but any floating IP addresses that are assigned to the instance might.

> | sec.lcm.009 | The Platform **must** be able to update newly instantiated, suspended, hibernated, migrated and restarted images with relevant IP addressing, including routing tables, route health information and whitelists/blacklists. |  | |

Reviewer 3: Floating IP addresses are a different thing; as written, this requirement seems to target the guest's allocated IP address.  It does not change across any of the suspend/hibernate/migrate/restart events.
Can we drop this requirement entirely and/or re-write it?

> | sec.lcm.009 | The Platform **must** be able to update newly instantiated, suspended, hibernated, migrated and restarted images with relevant IP addressing, including routing tables, route health information and whitelists/blacklists. |  | |

Author: @kedmison Deleted

> | sec.lcm.010 | The Platform **must** be able to update newly instantiated, suspended, hibernated, migrated and restarted images with relevant geolocation information |  | |

Reviewer 3: What would this geolocation info be used for, and how would this work in practice?

Author: @kedmison 
1. users may be constrained to run their workloads and save their data in certain geographies due to regulatory requirements and the VMs would need to be instantiated from images within these geographies
2. to support #1 above, need location and boundary control of OpenStack images

Reviewer 3: Thanks, the comment is very helpful.  The term 'geolocation' has some specific meaning that I think does not apply here.  
Is this needed if we have availability zone requirements, and permit operators to also assign geographical designations as needed?

Author: @kedmison I think so. Can reword to "The Platform **must** be able to update the tag of newly instantiated, suspended, hibernated, migrated and restarted images with relevant geolocation (geographical) information." 

Changed to suggested sentence above

| sec.lcm.011 | The Platform **must** log all changes to geolocation along with the mechanisms and sources of location information (i.e. GPS, IP block, and timing). |  | |

> | sec.lcm.012 | The Platform **must** implement Security life cycle management processes including proactively update and patch of all deployed software. | | |

Author: @kedmison LCM and Ops should cover this.

> | sec.lcm.012 | The Platform **must** implement Security life cycle management processes including proactively update and patch of all deployed software. | | |

Reviewer 3: The platform does not do updates for things inside the VNFs.  Perhaps we could clarify the 'all deployed software' to be 'all NFVI software'?

> | sec.lcm.012 | The Platform **must** implement Security life cycle management processes including proactively update and patch of all deployed software. | | |

Author: @kedmison Done -- Thanks

<a name="7.11.7"></a>
### 7.11.7. Monitoring and Security Audit

> The Platform is assumed to provide configurable alerting and notification capability and the operator is assumed to have automated systems, policies and procedures to act on alerts and notifications in a timely fashion. In the following the monitoring and logging capabilities can trigger alerts and notifications for appropriate action.

Reviewer 3: This section may represent significant scope creep
- full log monitoring, analysis, alerting
- incoming traffic analysis capabilities, DDOS protection
- detection of 'illegal' access
- automated response ("the platform must [...] take corrective security hardening measures")

This section should be re-visited to see how to discern log management & integrity functions that are required of the NFVI, from the requirements on a log management system, from the capabilities of Intrusion Detection System, from the capabilities of an Intrusion Prevention System, DDOS mitigation, and so on.

Author: @kedmison The LCM chapter has a section of Logging and monitoring (LMA). This section is providing requirements for some of what LMA is providing.

<a name="7.11.8"></a>
### 7.11.8. Compliance with Standards

| Ref | Requirement | Unit | Definition |
|---------|---------------|----------------|------------|
| sec.std.001 | The Cloud Operator **should** comply with Center for Internet Security CIS Controls ([https://www.cisecurity.org/](https://www.cisecurity.org/)) | | Center for Internet Security - [https://www.cisecurity.org/](https://www.cisecurity.org/) |
| | [Q: Are we going to verify compliance w Controls? If not, then why a “must” – but making it a “should” implies only guidance and not a control.] |  |  |
| sec.std.002 | The Cloud Operator, Platform and Workloads **should** follow the guidance in the CSA Security Guidance for Critical Areas of Focus in Cloud Computing (latest version) [https://cloudsecurityalliance.org/](https://cloudsecurityalliance.org/) |  | Cloud Security Alliance - [https://cloudsecurityalliance.org/](https://cloudsecurityalliance.org/) |
| sec.std.003 | The Platform and Workloads **should** follow the guidance in the OWASP Cheat Sheet Series (OCSS) https://github.com/OWASP/CheatSheetSeries |  | Open Web Application Security Project [https://www.owasp.org](https://www.owasp.org) |

> | sec.std.004 | The Cloud Operator, Platform and Workloads **should** ensure that their code is not vulnerable to the OWASP Top Ten Security Risks [https://owasp.org/www-project-top-ten/](https://owasp.org/www-project-top-ten/) |  |

Reviewer 3: This would also be a good addition to Appendix A: Guidelines for VNF vendors.

Author: @kedmison RM Appendix A or some other?

Reviewer 3: Yes, RM Appendix A.

Author: @kedmison Will open issue and add along with some of the image security requirements
